### DISCUSSION TITLE
Suggestions for editing Many to Many

### DISCUSSION THREAD

**Discussion by schwartzmj** [pocketbase/pocketbase#4076] [Q&A]
I'd love to use Pocketbase for a very simple app, but finding it not being practical to update a many-to-many relation.

Essentially the data model looks like:

- events (name, date)
- teams (name)
- scores (value, event_id, team_id)

Obviously editing Events and Teams is very simple in the Pocketbase admin panel. The issue comes when you're running an event and want to update scores for teams.

### Quickest solution, but not practical

By default, we essentially have a very large junction table. I suppose it works, but you have to go find your Event & Team row amongst many, many rows, click in and edit it, come back and find the next, and so on.

This method could _maybe_ work if it were easier to choose an Event to filter by, but as it stands you'd have to go find the event id and write `event.id = "some-id"` in the filter. You then click into each "score", update it, save, and ideally click "back" with the filter still applied so you can continue editing.

### What I've tested implementing

In our public-facing frontend, I made an `admin` route where you can login. I created some pages and forms and you can now edit the scores in a simpler way. Unfortunately you're now jumping between two different websites.

I could add some custom routes to Pocketbase so that you're not using two websites to manage the content, but with no control over the Pocketbase admin panel I cannot add a link to this custom route.

So if I go the route of managing content in my custom frontend, I'd essentially have to also set up pages to create and manage `teams` and `events` which is sort of defeating the purpose of using Pocketbase.

### In Conclusion

**Does anybody have any suggestions or ideas?** Maybe I could do something differently with how I'm structuring my collections or some other idea I haven't seen yet.

Maybe Pocketbase is just supposed to be a wrapper around a database that also provides an API and auth and I'm asking too much?

I love the simplicity and featureset of Pocketbase, but managing a relation like this seems impractical in the admin panel.

Thank you!

**Comment by ganigeorgiev** [comment]
> if it were easier to choose an Event to filter by...

You can filter by any field of the relation, not just the id, aka. `event.name = "example"`.

> I'd essentially have to also set up pages to create and manage teams and events in my custom frontend which is sort of defeating the purpose of using Pocketbase.

Note that the PocketBase Admin UI was intended to be used primarily as a dev/super-admin dashboard (eg. similar to the Firabase console). There are no roles or permission controls at the moment.

I don't have any suggestions but note that there are [plans](https://github.com/orgs/pocketbase/projects/2/views/1?pane=issue&itemId=32121831) in the near future to add option to access/edit indirect relations directly from the records panel that should make it slightly easier managing the references.

**Comment by jonshipman** [comment]
I've been implementing "View" collections for these kinds of problems. So in your case, create an `event_team_score` view and group on the event_id.

**Reply by schwartzmj** [reply]
Maybe I'm misunderstanding but View collections are uneditable and also don't provide a link to the relation to go edit it there.

**Reply by jonshipman** [reply]
It should if the column matches with a collection it should auto build a relation

**Reply by ganigeorgiev** [reply]
@schwartzmj is right and View collections are read-only and relations can't be managed/edited from there.

**Reply by jonshipman** [reply]
Sorry, misunderstood. I was saying the relation should be there if you're doing a join on a table.

e.g. 
```
SELECT
		l.id AS id,
		l.id AS lsa,
		jp.job AS job,
FROM
		lsa l
INNER JOIN jobPhone jp ON l.consumerPhoneNumber = jp.number
INNER JOIN job j ON jp.job = j.id
```

Will create a relation to the lsa table from the lsa column and the job table from the job column. From the js-sdk they can be triggered to expand with 'job,lsa'.

You will then need to have a custom application to modify both entities from one record.
